pkg/volume/cephfs/cephfs.go

```diff
@@ -89,62 +89,70 @@ func (plugin *cephfsPlugin) GetAccessModes() []v1.PersistentVolumeAccessMode {
 }
 
 func (plugin *cephfsPlugin) NewMounter(spec *volume.Spec, pod *v1.Pod, _ volume.VolumeOptions) (volume.Mounter, error) {
-	cephvs, _, err := getVolumeSource(spec)
+	_, _, _, _, localSecretRef, secretRef, _, err := getVolumeSource(spec)
 	if err != nil {
 		return nil, err
 	}
+
+	secretNs := pod.Namespace
+	secretName := ""
 	secret := ""
-	if cephvs.SecretRef != nil {
-		kubeClient := plugin.host.GetKubeClient()
-		if kubeClient == nil {
-			return nil, fmt.Errorf("Cannot get kube client")
+	if localSecretRef != nil {
+		secretName = localSecretRef.Name
+	} else if secretRef != nil {
+		if len(secretRef.Namespace) != 0 {
+			secretNs = secretRef.Namespace
 		}
+		secretName = secretRef.Name
+	}
+	kubeClient := plugin.host.GetKubeClient()
+	if kubeClient == nil {
+		return nil, fmt.Errorf("Cannot get kube client")
+	}
 
-		secretName, err := kubeClient.Core().Secrets(pod.Namespace).Get(cephvs.SecretRef.Name, metav1.GetOptions{})
-		if err != nil {
-			err = fmt.Errorf("Couldn't get secret %v/%v err: %v", pod.Namespace, cephvs.SecretRef, err)
-			return nil, err
-		}
-		for name, data := range secretName.Data {
-			secret = string(data)
-			glog.V(4).Infof("found ceph secret info: %s", name)
-		}
+	secrets, err := kubeClient.Core().Secrets(secretNs).Get(secretName, metav1.GetOptions{})
```

Comment:
should probably be using `plugin.host.GetSecretFunc()` to make use of the secret manager in the kubelet that has already fetched this secret